src/components/tasks/task-connections-table.tsx

```diff
@@ -0,0 +1,272 @@
+      case TaskConnectionType.FinishToStart:
+        return sourceTask.attachedTask.endDate > targetTask.attachedTask.startDate
+          ? t("newMilestoneTaskDialog.taskConnectionsTable.taskTypeCheck.sourceEndAfterTargetStartWarning")
+          : null;
+      default:
+        return null;
+    }
+  };
+
+  /**
+   * Recalculate overall validity
+   */
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    const allConnections = [...existingTaskConnections, ...newTaskConnections];
+    const validationErrors = allConnections.map(connection =>
+      determineIfTaskTypeIsAllowed(
+        { ...connection, attachedTask: milestoneTasks.find((task) => task.id === connection.attachedTask?.id) },
+        { ...connection, attachedTask: currentTask },
+        connection.type
+      )
+    );
+    setTaskConnectionsValid(!validationErrors.some(error => error !== null));
+  }, [existingTaskConnections, newTaskConnections, milestoneTasks, currentTask]);
+
+  /**
+   * Render existing task connection table rows
+   */
+  const renderExistingTaskConnectionTableRows = () => {
+    return existingTaskConnections.map(connection => (
+      <TableRow key={connection.connectionId}>
+        <TableCell>
+          <TextField value={connection.hierarchy} disabled />
+        </TableCell>
+        <TableCell>
+          <TextField
+            fullWidth
+            select
+            value={connection.type}
+            onChange={(e) => handleEditConnection(connection.connectionId, "type", e.target.value as TaskConnectionType)}
+          >
+            {Object.values(TaskConnectionType).map((type) => (
+              <MenuItem key={type} value={type}>
+                {type}
+              </MenuItem>
+            ))}
+          </TextField>
+          <div style={{ color: 'red', fontSize: '0.75rem' }}>
+            {determineIfTaskTypeIsAllowed(
+              { ...connection, attachedTask: milestoneTasks.find((task) => task.id === connection.attachedTask?.id) },
+              { ...connection, attachedTask: currentTask },
+              connection.type
+            )}
+          </div>
+        </TableCell>
+        <TableCell>
+          <TextField fullWidth value={connection.attachedTask?.name ?? ""} disabled />
+        </TableCell>
+        <TableCell>
+          <TextField fullWidth value={connection.attachedTask?.status ?? ""} disabled />
+        </TableCell>
+        <TableCell>
+          <IconButton onClick={() => removeExistingTaskConnecitonRow(connection.connectionId)}>
+            <DeleteIcon />
+          </IconButton>
+        </TableCell>
+      </TableRow>
+    ));
+  };
+
+  /**
+   * Render new task connection table rows
+   */
+  const renderNewTaskConnectionTableRows = () => {
+    return newTaskConnections.map((connection, index) => (
+      // biome-ignore lint/suspicious/noArrayIndexKey: With new task connection rows we only have the index as key
+      <TableRow key={index}>
```

Comment:
Use connection.connectionId as the key?

Reply:
The problem here is that connection doesn't have a connectionId before it is created. And it feels like assigning a dummy connection Id as some random UUID is a worse idea than just using an index. Or is it?

Reply:
Now refactored to fix it